docs/function-schemas.md

```diff
@@ -319,11 +331,29 @@ Generating multi-arity functions:
 ; => -2326
 ```
 
+### Polymorphic Functions
+
+A polymorphic function using `m/all` is generatively tested by instantiating schema variables with small schemas.
+
+For example, the polymorphic identity schema
+
+```clojure
+(m/all [a] [:=> [:cat a] a])
+```
+
+is generatively tested with schemas like
+
+```clojure
+[:=> [:cat :nil] :nil]
+[:=> [:cat [:enum 50]] [:enum 50]]
+[:=> [:cat [:enum 5333344553]] [:enum 5333344553]]
```

Comment:
Should you have three different schemas here? I know the two enums are technically different, but maybe `:nil`, `[:enum 50]` and `[:map-of :keyword :any]` or something would better clarify what's happening.

Reply:
Currently it really is this simple, more like:

```clojure
[:fn {:gen/return nil} (fn [r] (= nil r))]
[:fn {:gen/return 50} (fn [r] (= 50 r))]
[:fn {:gen/return 5333344553} (fn [r] (= 5333344553 r))]
```

Reply:
Oh interesting, okay.

Reply:
To go further we'll need to think about what it means to shrink a schema. There are ideas in the spec impl of this, but this is the minimal proof-of-concept.

Reply:
I will elaborate on exactly what happens in the docs though, thanks for the feedback.

Reply:
Found the reason I needed such a convoluted representation in the first place: https://github.com/metosin/malli/issues/1070

Reply:
> I will elaborate on exactly what happens in the docs though, thanks for the feedback.

Done.